### PR TITLE
test(providers/microsfot): add system test for AzureContainerVolumeHook and AzureContainerRegistryHook

### PR DESCRIPTION
**Pull request by Lee-W**
During my work on https://github.com/apache/airflow/pull/33825 and https://github.com/apache/airflow/pull/33822, I discovered that there are no system tests for AzureContainerVolumeHook and AzureContainerRegistryHook. Is there any setup I need to do for these tests, or are they sufficient as is?

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
> During my work on #33825 and #33822, I discovered that there are no system tests for AzureContainerVolumeHook and AzureContainerRegistryHook. Is there any setup I need to do for these tests, or are they sufficient as is?

I guess Azure peopls are missing-in-action so you are on your own :). Generally you should be able to run those example_dags with real Azure account.

File: tests/system/providers/microsoft/azure/example_azure_container_instances.py [hunk 1]
```diff
@@ -52,7 +55,39 @@
...
+        image="hello-world",
+        region="WestUS2",
+        environment_variables={},
+        volumes=[
+            AzureContainerVolumeHook().get_file_volume(
+                mount_name="mountname",
+                share_name=AZURE_VOLUME_SHARE_NAME,
+                storage_account_name=AZURE_STORAGE_ACOOUNT,
+            )
+        ],
```
Comment (Taragolis) [inline tests/system/providers/microsoft/azure/example_azure_container_instances.py]: This one spawn DB connection during DAG parsing that the reason why test failed.

In general for this cases better use taskflow operator which returns required value, however volumes are not templated fields, so maybe better make it templated first.

Some samples from other providers

https://github.com/apache/airflow/blob/6ef80e8be178e0ab8d119270a28b23d0bf47ed62/tests/system/providers/amazon/aws/example_batch.py#L60-L83

https://github.com/apache/airflow/blob/d2c0bbebacd1c58d357d647707a1bdb0c5b3b425/tests/system/providers/amazon/aws/example_ec2.py#L42-L60

**Comment by Lee-W** [comment]
> > During my work on #33825 and #33822, I discovered that there are no system tests for AzureContainerVolumeHook and AzureContainerRegistryHook. Is there any setup I need to do for these tests, or are they sufficient as is?
> 
> I guess Azure peopls are missing-in-action so you are on your own :). Generally you should be able to run those example_dags with real Azure account.

Yep, these example_dags are good with our Azure account. But I thought these tests would be run on the main branch so there would be some connection setup needed. Or do I misunderstand the system tests?

File: tests/system/providers/microsoft/azure/example_azure_container_instances.py (hunk 1 above)
Comment (Lee-W) [reply tests/system/providers/microsoft/azure/example_azure_container_instances.py]: Got it. Thanks for helping out. Let me turn this to draft and work on that first.

Comment (Lee-W) [reply tests/system/providers/microsoft/azure/example_azure_container_instances.py]: I just found out I use `AzureContainerVolumeHook` the wrong way. Just update it with the correct usage

**Comment by potiuk** [comment]
> Yep, these example_dags are good with our Azure account. But I thought these tests would be run on the main branch so there would be some connection setup needed. Or do I misunderstand the system tests?

Nope. System tests are run by:

a) individual contributors who have account to connect to the 3rd-party cloud (if they choose to)
b) our stakeholders who take care of them (Amazon currently, Google in progress). What we have access to is the system test dashboard they provide (Amazon: https://aws-mwaa.github.io/open-source/system-tests/dashboard.html )  which we can review when releasing

This is all part of I think one of the latest AIP we have [AIP-4](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=93324355)  later extended bu [AIP-47](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-47+New+design+of+Airflow+System+Tests)

**Review by potiuk — APPROVED** [review]

**Comment by Lee-W** [comment]
> > Yep, these example_dags are good with our Azure account. But I thought these tests would be run on the main branch so there would be some connection setup needed. Or do I misunderstand the system tests?
> 
> Nope. System tests are run by:
> 
> a) individual contributors who have account to connect to the 3rd-party cloud (if they choose to) b) our stakeholders who take care of them (Amazon currently, Google in progress). What we have access to is the system test dashboard they provide (Amazon: https://aws-mwaa.github.io/open-source/system-tests/dashboard.html ) which we can review when releasing
> 
> This is all part of I think one of the latest AIP we have [AIP-4](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=93324355) later extended bu [AIP-47](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-47+New+design+of+Airflow+System+Tests)

Got it. Thanks for the detailed illustration!

**Pull request merged**